### ISSUE TITLE
requests.exceptions.Timeout: UnixHTTPConnectionPool

### ISSUE THREAD

**Issue by dimaip** [docker/compose#1045]
Hi! I often get the following error on our Centos 6.5 server when executing `docker-compose up -d`:
```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 27, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 59, in perform_command
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 155, in logs
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 205, in containers
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 519, in containers
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 81, in _get
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 395, in get
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 383, in request
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 486, in send
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.adapters", line 387, in send
requests.exceptions.Timeout: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
```

Any idea what's happening?

**Comment by dnephin** [comment]
Are there many things running on this host hitting the dockerd?  I've only seen errors like this one while running many builds in parallel,  and the problem ended up being https://github.com/docker/docker/issues/9656

**Comment by dimaip** [comment]
No, not at all.
I get around this error by this:
```
docker-compose stop
docker-compose rm
docker-compose up -d
```
Could this be something related to CentOS 6.5? On Ubuntu I experience nothing like this.

**Comment by thomgit** [comment]
Same error with centos7 . The above workaround does not help. debugging...

**Comment by psychowico** [comment]
I have it almost always when I do docker compose first start, after my computer reboot.

**Comment by actionless** [comment]
i can reproduce it when building compose project which have more than 2-3 docker images on a server with slow disc i/o

**Comment by actionless** [comment]
seems to be the same issue:
https://github.com/docker/compose/issues/812

**Comment by ryan-schroeder** [comment]
+1 experiencing on lubuntu 15 VM, nothing else touching docker, fresh reboot.  Huh!

**Comment by actionless** [comment]
@ryan-schroeder i seen in the changelog what docker-compose v1.2 received timeout option, can you try with it?

**Comment by amosjyng** [comment]
Experienced this on lubuntu VM as well.

**Comment by kamermans** [comment]
I see this issue as well, perhaps 50% of the time when I launch `c3.xlarge` SSD-backed instances in AWS and put `docker-compose up -d` in `rc.local`.  I'm considering running `sleep 10` in `rc.local` to let the CPU settle down a bit before it dives into Docker.  I'm bringing up 7 containers via docker-compose.  I'm also curious about the timeout option.  Here's my output from an instance that was started from an autoscaling group:

```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 27, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 59, in perform_command
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 464, in up
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 208, in up
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 222, in recreate_containers
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 247, in recreate_container
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 36, in create
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 425, in create_container
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 432, in create_container_from_config
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 119, in _post_json
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 79, in _post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 425, in post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 383, in request
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 486, in send
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.adapters", line 387, in send
requests.exceptions.Timeout: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
```

I'm running `docker-compose 1.2.0`.

**Comment by actionless** [comment]
> I'm also curious about the timeout option.
Timeout option not working for you? After it was added and i set timeout 5 times larger I am not experiencing the issue no more.

**Comment by kamermans** [comment]
I am still seeing this issue but I do notice that it only happens when I launch a new instance from an AMI (manually or via autoscaling).  I think when you launch a new instance there is some really bad I/O for a while - maybe the AMI is being copied and so at launch you're really in a snapshot or something, but in any case, it's not working on launch from an AMI but it works OK when I just reboot an instance.

```
$ /usr/local/bin/docker-compose up -d --timeout 180
Recreating compose_rsyslog_1...
Recreating compose_app_1...
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 27, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 59, in perform_command
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 464, in up
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 208, in up
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 222, in recreate_containers
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 247, in recreate_container
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 36, in create
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 425, in create_container
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 432, in create_container_from_config
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 119, in _post_json
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 79, in _post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 425, in post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 383, in request
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 486, in send
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.adapters", line 387, in send
requests.exceptions.Timeout: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
```

Notice that despite my timeout of `180`, the error still says `read timeout=60`.  Strange.

**Comment by actionless** [comment]
@kamermans 
Can you try with 600? "My" number was about 240.

**Comment by kamermans** [comment]
I'm using v1.2.0.  I've switched from aufs to device mapper to reduce the copy burden on run, and I've put the volumes and temp dirs on other disks.  Now the whole setup comes up in 30 sec or so.  I've pulled compose out for now as I need to get this into production.

Btw, here is a script to put docker's volumes and temp dirs on ephemeral storage at EC2:
https://gist.github.com/kamermans/176c190670a163b1147e

**Comment by schmunk42** [comment]
Hey guys, you can set the read timeout with

    export DOCKER_CLIENT_TIMEOUT=120

Found [here](https://github.com/docker/compose/blob/9a0bb325f2d1203b7aac915c3bfca4347cc93489/compose/cli/docker_client.py#L34) via [code-search](https://github.com/docker/compose/search?utf8=✓&q=timeout). Tested in our GitLab CI, which had the exact same problems, now running like a charm even with several concurrent builds.

Is this in the docs? [Don't think so](https://github.com/docker/compose/search?utf8=✓&q=DOCKER_CLIENT_TIMEOUT) :)

**Comment by rarkins** [comment]
@schmunk42 Thanks for sharing this possible solution. I've tried the 120s timeout for a few days and unfortunately not found any difference. I still get plenty of timeouts at 120s now instead of 60. Example:

```
$ make
Killing app_logspout_1...
Killing app_seed_1...
Killing app_webpack_1...
Killing app_nginx_1...
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 31, in main
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 21, in sys_dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 27, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.docopt_command", line 24, in dispatch
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.command", line 59, in perform_command
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.main", line 144, in kill
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.project", line 182, in kill
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.service", line 110, in kill
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.container", line 127, in kill
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 605, in kill
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.client", line 79, in _post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 425, in post
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 383, in request
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.sessions", line 486, in send
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.adapters", line 387, in send
requests.exceptions.Timeout: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=120)
make: *** [stop] Error 255
$
```

So I'm guessing that whatever problem I'm seeing is not solved by increasing the timeout, no matter how big.

**Comment by schmunk42** [comment]
@rarkins Could you posts the CPU and Memory of that machine? Looks to me like there's still an issues with limited resources. Are there other processes running on the Docker host also?

**Comment by rarkins** [comment]
Docker and unison are running but the latter shouldn't in theory be transferring anything during that time. Is taking a snapshot of "top" during that timeout sufficient or is there a more reliable way to record/monitor that you'd recommend?

**Comment by schmunk42** [comment]
I just (recently :smile:) "discovered" `docker stats`, which may also help here.
You also also try to connecting manually to `docker` during the startup phase and see if you've the same problems or if this is an issue with the containers.

**Comment by rarkins** [comment]
This container was in the middle of a timeout when I captured this:
```
CONTAINER           CPU %               MEM USAGE/LIMIT     MEM %               NET I/O
app_nginx_1         0.00%               6.816 MiB/490 MiB   1.39%               3.155 KiB/2.711 KiB
```

Running `top` on the system revealed it running at 3.7%.

After it timed out (120 seconds), running the same command again saw it killed in an instant. i.e. this is not a CPU or memory-intensive task.

**Comment by rarkins** [comment]
Given this observation it's actually my preference to *reduce* the timeout so at least it fails faster!

**Comment by yuemin-li** [comment]
same problem happened in OSX when I try to run some app using existing container after 10 hours.
```
Traceback (most recent call last):
  File "/usr/local/Cellar/fig/1.3.0/libexec/bin/docker-compose", line 9, in <module>
    load_entry_point('docker-compose==1.3.0', 'console_scripts', 'docker-compose')()
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/cli/main.py", line 31, in main
    command.sys_dispatch()
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 21, in sys_dispatch
    self.dispatch(sys.argv[1:], None)
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/cli/command.py", line 34, in dispatch
    super(Command, self).dispatch(*args, **kwargs)
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/cli/docopt_command.py", line 24, in dispatch
    self.perform_command(*self.parse(argv, global_options))
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/cli/command.py", line 66, in perform_command
    handler(project, command_options)
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/cli/main.py", line 399, in stop
    project.stop(service_names=options['SERVICE'], **params)
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/project.py", line 191, in stop
    service.stop(**options)
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/service.py", line 130, in stop
    c.stop(**options)
  File "/usr/local/Cellar/fig/1.3.0/libexec/lib/python2.7/site-packages/compose/container.py", line 139, in stop
    return self.client.stop(self.id, **options)
  File "/usr/local/Cellar/fig/1.3.0/libexec/vendor/lib/python2.7/site-packages/docker/utils/decorators.py", line 15, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/usr/local/Cellar/fig/1.3.0/libexec/vendor/lib/python2.7/site-packages/docker/client.py", line 1097, in stop
    timeout=(timeout + self.timeout))
  File "/usr/local/Cellar/fig/1.3.0/libexec/vendor/lib/python2.7/site-packages/docker/client.py", line 103, in _post
    return self.post(url, **self._set_request_timeout(kwargs))
  File "/usr/local/Cellar/fig/1.3.0/libexec/vendor/lib/python2.7/site-packages/requests/sessions.py", line 508, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/local/Cellar/fig/1.3.0/libexec/vendor/lib/python2.7/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/Cellar/fig/1.3.0/libexec/vendor/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/Cellar/fig/1.3.0/libexec/vendor/lib/python2.7/site-packages/requests/adapters.py", line 433, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='192.168.59.103', port=2376): Read timed out. (read timeout=70)
```
docker-stop doesn't help in this case, cause it still goes to a timeout. Only 4 containers in the docker-compose.yml file.

**Comment by dnephin** [comment]
There are a bunch of these ConnectionTimeout issues, I'm consolidating them all into  #1923.

There isn't much we can do about this from the client, except for adding a more helpful error message.

**Comment by ifeatu** [comment]
+1 This is still a problem.

**Comment by nirmalpathak** [comment]
+1 This is still a problem with 
# docker version
Client version: 1.7.1

# docker-compose --version
docker-compose version: 1.4.0

**Comment by wicope** [comment]
host - vagrant - docker

-- docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64

-- docker-compose --version
docker-compose version: 1.4.0

-- docker-compose logs xyxy
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/code/build/docker-compose/out00-PYZ.pyz/threading", line 810, in __bootstrap_inner
  File "/code/build/docker-compose/out00-PYZ.pyz/threading", line 763, in run
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.multiplexer", line 41, in _enqueue_output
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.log_printer", line 59, in _make_log_generator
  File "/code/build/docker-compose/out00-PYZ.pyz/compose.cli.utils", line 100, in split_buffer
  File "/code/build/docker-compose/out00-PYZ.pyz/docker.clientbase", line 238, in _stream_raw_result
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.utils", line 332, in stream_decode_response_unicode
  File "/code/build/docker-compose/out00-PYZ.pyz/requests.models", line 680, in generate
ConnectionError: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out.

**Comment by Krym** [comment]
+1 This is still a problem.

Running ubuntu 14.04 on Azure
docker version 1.7.1
docker-compose verxion 1.4.2

**Comment by vitalyisaev2** [comment]
+1

Centos 6.5
Docker 1.6.0
Docker-py 1.2.3

**Comment by dnephin** [comment]
If you get a timeout on the client (docker-compose is a client) it's almost always an issue with the server. 60 seconds should be more than enough time to wait for a response in most cases.  Since you're running such an old version of docker, it's probably worth upgrading docker.

We've included a better error message in the 1.5.0 release, but there isn't anything else we can do on the compose side.

**Comment by Ghjnut** [comment]
still seeing this 
```
$ docker version
Client:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:10:10 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:10:10 UTC 2015
 OS/Arch:      linux/amd64

**Comment by lcacciagioni** [comment]
Same here
```
docker version
Client:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   f4bf5c7
 Built:        Mon Oct 12 05:33:35 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   f4bf5c7
 Built:        Mon Oct 12 05:33:35 UTC 2015
 OS/Arch:      linux/amd64
```

Please re open this issue

**Comment by dnephin** [comment]
Please include the Compose version you are using (not just the docker version).

Timeouts are not an issue with the client (Compose). All we can do from compose is provide a better error message (which we've done in 1.5.0).

**Comment by aberope** [comment]
$ docker-compose --verbose down
compose.config.config.find: Using configuration files: ./docker-compose.yml
docker.auth.auth.load_config: File doesn't exist
compose.cli.command.get_client: docker-compose version 1.7.0, build 0d7bf73
docker-py version: 1.8.0
CPython version: 2.7.6
OpenSSL version: OpenSSL 1.0.1f 6 Jan 2014
[...]
ERROR: for logstash  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=70)
ERROR: An HTTP request took too long to complete. Retry with --verbose to obtain debug information.
If you encounter this issue regularly because of slow network conditions, consider setting COMPOSE_HTTP_TIMEOUT to a higher value (current value: 60).

**Comment by wicope** [comment]
Hi aberope.

Try:
```
COMPOSE_HTTP_TIMEOUT 600 docker-compose --verbose down
```
where you can change 600 for your needs

**Comment by aberope** [comment]
Thanks wicope.

$ docker --version
Docker version 1.11.1, build 5604cbe

I rebooted the server and hanged up on reboot. I had to hardware reset...
I tried to restart docker' service.
Seems that the server had bottleneck, it was some process(es) in background making the system delay and go slow. 
I guess were the docker, but I am not sure because we just installed two new things, after a stable running of the server:
1. we orchestate containers with docker-compose (new)
2. installed sysdig monitoring system (new)
I can't be sure for now, about the reason of the error, but seems that docker was bugged or sysdig bugged the system and docker was slow.

If this happens again I'll try this command.

**Comment by jlongie** [comment]
Having the same issue with the following steps on a Jenkins CI build. The machine has been restarted, versions upgraded, after each build and run there is a cleanup script to stop/remove containers.

Building the image(s)
```sh
export COMPOSE_HTTP_TIMEOUT=120
cat compose.yml | envsubst | docker-compose -f - build app
```

Running the image(s)
```sh
export COMPOSE_HTTP_TIMEOUT=120
cat compose.yml | envsubst | docker-compose -f - up
```

Sample compose file:
```sh
version: '2'

services:
  mongodb:
    image: mongo:3.2.4
    ports:
      - "27017:27017"
    mem_limit: 1024m
    cpu_shares: 10
    networks:
      ntwk:
        aliases:
          - dbhost

  app:
    build: .
    image: domain/app:${DOCKER_IMAGE_TAG}
    ports:
      - "9000:9000"
    environment:
      - MONGODB_HOST=mongodb
    mem_limit: 7168m
    cpu_shares: 50
    depends_on:
      - mongodb
    networks:
    ntwk:
        aliases:
          - main-app

networks:
  ntwk:
```

Image sizes:
```sh
mongodb 600mb
app: 2.5gb
```

Versions
```sh
Docker version 1.11.1, build 5604cbe
docker-compose version 1.7.1, build 0a9ab35
docker-machine version 0.7.0, build a650a40
```

The following error message appears 50% of the time - this is happening on our Jenkins CI environment and also on our mac's!
```sh
08:51:51 ERROR: for app  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=130)
08:51:51 An HTTP request took too long to complete. Retry with --verbose to obtain debug information.
08:51:51 If you encounter this issue regularly because of slow network conditions, consider setting COMPOSE_HTTP_TIMEOUT to a higher value (current value: 60).
08:51:51 Build step 'Execute shell' marked build as failure
```

**Comment by wicope** [comment]
Hi jlongie. 
It is some notes, is what I would do if things do not improve.
To remove complexity is best to leave only the essentials: docker and docker-compose.
Run docker console commands and see the result. (pass from format yml to docker command).
Run docker-compose console commands and see the result.
You can see the output of both commands and get conclusions.
Later, after work, you include jenkins. Regards

**Comment by packageman** [comment]
@schmunk42 The env has chagned to `COMPOSE_HTTP_TIMEOUT` since version 1.8.0 and above.

**Comment by rydente** [comment]
Just wanted to let you all know, this is still a thing which happens, and didn't happen until the Community Edition of Docker got released:

`ERROR: for app_legacy_1  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)`
`ERROR: for app_partner4-mobile_1  Cannot create container for service partner4-mobile: Device is Busy`
`WARNING: Connection pool is full, discarding connection: localhost`
```
ERROR: An HTTP request took too long to complete. Retry with --verbose to obtain debug information.
If you encounter this issue regularly because of slow network conditions, consider setting COMPOSE_HTTP_TIMEOUT to a higher value (current value: 60).
```

(Note: you can't actually use a `--verbose` option with Docker Compose.)

`docker info`:
```
[dakhath@sidabras ~]$ docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   3dfb8343
 Built:        Wed Jul 26 18:03:33 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87617
 Built:        Wed Jul 26 20:03:39 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

```
[dakhath@sidabras app]$ docker-compose -v
docker-compose version 1.15.0, build unknown
[dakhath@sidabras app]$ 
```

I had to physically use this command to make things run locally (`alias compose='docker-compose'
`):

```
compose down && COMPOSE_HTTP_TIMEOUT=600 compose up -d && compose logs -f
```

Letting you know this is still an issue and has been since at least March.

**Comment by goodspark** [comment]
Just saw this today as well.

**Comment by brainstorm** [comment]
This is still a issue for us as well, it shows up every now and then, making packer deployments hard to consolidate: 

https://github.com/umccr/infrastructure/blob/master/packer/stackstorm-ami/bootstrap-stackstorm.sh#L25

/cc @reisingerf

**Comment by gecube** [comment]
I faced the same issue yesterday and was shocked. I have two identical servers. The only difference is that on affected server I am using `docker-compose version 1.23.2, build 1110ad01` installed as single binary from official site. It is interesting that previous issue was with the docker daemon recreating networks in cycle and I fixed it by total uninstalling NetworkManager from system. Generally speaking the system is working and I will proceed further with examination of the configuration and the monitoring of system.
